contracts/voting/dao-voting-cw4/src/contract.rs

```diff
@@ -164,12 +217,48 @@ pub fn query_total_power_at_height(deps: Deps, env: Env, height: Option<u64>) ->
 }
 
 pub fn query_info(deps: Deps) -> StdResult<Binary> {
-    let info = cw2::get_contract_version(deps.storage)?;
+    let info = get_contract_version(deps.storage)?;
     to_json_binary(&dao_interface::voting::InfoResponse { info })
 }
 
+pub fn query_is_active(deps: Deps) -> StdResult<Binary> {
+    let active_threshold = ACTIVE_THRESHOLD.may_load(deps.storage)?;
+
+    match active_threshold {
+        Some(ActiveThreshold::AbsoluteCount { count }) => {
+            let group_contract = GROUP_CONTRACT.load(deps.storage)?;
+            let total_weight: TotalWeightResponse = deps.querier.query_wasm_smart(
+                &group_contract,
+                &cw4_group::msg::QueryMsg::TotalWeight { at_height: None },
+            )?;
+            to_json_binary(&IsActiveResponse {
+                active: total_weight.weight >= count.u128() as u64,
+            })
+        }
+        Some(ActiveThreshold::Percentage { percent }) => {
+            let group_contract = GROUP_CONTRACT.load(deps.storage)?;
+            let total_weight: TotalWeightResponse = deps.querier.query_wasm_smart(
+                &group_contract,
+                &cw4_group::msg::QueryMsg::TotalWeight { at_height: None },
+            )?;
+            let percentage_base = Uint256::from(10u64).pow(percent.decimal_places()); // Ensure correct power base for scaling
+            let required_weight = Uint256::from(total_weight.weight)
+                .multiply_ratio(percent.atomics(), percentage_base); // Correct ratio calculation
+
+            to_json_binary(&IsActiveResponse {
+                active: Uint256::from(total_weight.weight) >= required_weight,
+            })
+        }
```

Comment:
`ActiveThreshold::Percent` should NOT be used with this contract. Do you see that you're just comparing total_weight.weight` to `total_weight.weight`?

Reply:
Please read my earlier comments about not allowing `ActiveThreshold::Percent`.